static/js/message_edit.js

```diff
@@ -176,15 +181,46 @@ exports.save = function (row, from_topic_edited_only) {
         message_edit.end(row);
         return;
     }
+
+    if (changed && !topic_changed) {
+        currently_echoing_messages[message_id] = message.raw_content;
+        message.locally_echoed = true;
+        echo.edit_locally(message, new_content);
+        row = current_msg_list.get_row(message_id);
+        message_edit.end(row);
+    }
+
     channel.patch({
         url: '/json/messages/' + message.id,
         data: request,
         success: function () {
             var spinner = row.find(".topic_edit_spinner");
             loading.destroy_indicator(spinner);
+
+            message.locally_echoed = false;
+            delete currently_echoing_messages[message.id];
         },
         error: function (xhr) {
             if (msg_list === current_msg_list) {
+                message_id = rows.id(row);
+
+                var echoed_message = message_store.get(message_id);
+
+                var old_content = currently_echoing_messages[message_id];
+                new_content = echoed_message.raw_content;
+
+                echoed_message.locally_echoed = false;
+
+                echo.edit_locally(echoed_message, old_content);
+
+                row = current_msg_list.get_row(message_id);
+
+                delete currently_echoing_messages[message_id];
+
+                if (!message_edit.is_editing(message_id)) {
+                    exports.start_edit_maintaining_scroll(row, new_content);
```

Comment:
I believe this should just be a call to `start_edit_maintaining_scroll`, not as a property of `exports`.

Reply:
Thanks, I made this change in my most recent commit and it looks to be working now. Can you test on your end if you're getting the same error handling as this? https://gfycat.com/vapidmealyblackrhino

Reply:
Ended up refactoring based on feedback from the dev-chat.